### DISCUSSION TITLE
How to query for _count in a regular query(not aggregate query)

### DISCUSSION THREAD

**Discussion by omar-dulaimi** [MichalLytek/typegraphql-prisma#142] [Q&A] [answered]
I have made the changes to the resolvers as suggested by generated resolvers:

```
@TypeGraphQL.Query(_returns => [Cat], {
    nullable: false,
})
async cats(
    @TypeGraphQL.Ctx() ctx: Context,
    @TypeGraphQL.Info() info: GraphQLResolveInfo,
    @TypeGraphQL.Args() args: FindManyCatArgs,
): Promise<Cat[]> {
    const { _count } = transformFields(graphqlFields(info as any));
    return getPrismaFromContext(ctx).cat.findMany({
        ...args,
        ...(_count && transformCountFieldIntoSelectRelationsCount(_count)),
    });
}
```

From frontend, I expect this to work:

```
query {
   cats {
      id
      _count {
         id
      } 
   }
}

```

I currently get 400 errors. So Am I using this correctly?

**Comment by omar-dulaimi** [comment, marked as answer]
So it's a preview feature, and enabled by adding this `selectRelationCount` to previewFeatures array in client generator in prisma schema.

But if it's a preview feature, then why did all the resolvers change? Isn't that unnecessary?

@MichalLytek

**Reply by MichalLytek** [reply]
@omar-dulaimi Maybe I forgot to hide that field from schema when preview feature is not enable 👀

**Reply by MichalLytek** [reply]
@omar-dulaimi Fixed by cae2242 🔒
